riak/datadog_checks/riak/riak.py

```diff
@@ -247,15 +248,7 @@ def check(self, _):
             )
             raise
 
-        if resp.status != 200:
-            self.service_check(
-                self.SERVICE_CHECK_NAME,
-                AgentCheck.CRITICAL,
-                tags=self.service_check_tags,
-                message="Unexpected status of %s when fetching Riak stats, response: %s" % (resp.status, content),
-            )
-
-        stats = json.loads(content)
+        stats = json.loads(r.content)
```

Comment:
Let's parse the json in the try/except block. That way if the json data is invalid, a CRITICAL service check will be submitted.